### DISCUSSION TITLE
Does array contain any of several values

### DISCUSSION THREAD

**Discussion by smach** [uwdata/arquero#314] [Q&A] [answered]
I am trying to use Arquero with an Observable Inputs.select where multiple values can be selected, such as 

```
sourceFilterValue = Inputs.select(available_source_array, { label: "Select source: ", value: "All", multiple: true 
})
```

For an Input.select where only one value can be selected, I know that I can use Arquero to filter with 

```
filteredTable = myTable
  .params({ sourceFilterValue })
  .filter((d) =>
     sourceFilterValue !== "All" ? d.SourceColumn === sourceFilterValue : true)
```

What would I do to find "all table rows where SourceColumn is one of my sourceFilterValues"? (the Arquero equivalent of R's `SourceColumn %in% sourceFilterValue` ) Trying to turn the sourceFilterValue array into a regular expression with `sourceFilterValue.join("|")` and using `op.match()` didn't work.

Update: I guess i'll use a vanilla JS solution instead of trying to create a regex for op.match.

Thanks.

**Comment by smach** [comment, marked as answer]
If anyone else is wondering, I used conventional JavaScript instead of a table expression since I didn't see a way to use Arquero specifically for this:

`.filter(aq.escape((d) => sourceFilterValue[0] !== "All" ? sourceFilteringValue.includes(d.SourceColumn) : true) )`

**Comment by bmschmidt** [comment]
FWIW, I always just make the filtering values into a new table and do an inner join for this kind of thing:

```js
filteredTable = myTable
    .join(aq.table({SourceColumn: sourceFilterValue[0] === "All" ? available_source_array : sourceFilterValue}))
```
